php/blocks/controls/class-rich-text.php

```diff
@@ -51,6 +51,12 @@ public function register_settings() {
 	 */
 	public function validate( $value, $echo ) {
 		unset( $echo );
+
+		// For this value, wpautop() returns '</p>', so simply return the value.
+		if ( '<p></p>' === $value ) {
```

Comment:
Something like this might handle more weird edge cases (such as `<span></span>` or `<p><strong></strong></p>`):

```PHP
if ( empty( esc_html( $value ) ) ) {
    return $value;
}
// or…
if ( '' === esc_html( $value ) ) {
    return $value;
}
```

Reply:
Ah, interesting point.

I'm not sure if there will be a case where `empty( esc_html( $value ) )`, or `'' === esc_html( $value )`, though maybe I just haven't seen it. 

It looks like even if there's no value entered in the Rich Text control, its value is `<p></p>`

Reply:
I could be missing something here.
- If the Rich Text control value is empty, the value is set to `<p></p>`
- The expected value would be an empty string, right?
- The logic above says "After removing all the HTML, if I'm left with an empty string, return an empty string"

Reply:
Hi @lukecarbis,
All good, those are good points.

>If the Rich Text control value is empty, the value is set to `<p></p>`

Right 😄 

>The expected value would be an empty string, right?

Good question, maybe 😄 

The way I thought of this was like the Paragraph block, which also uses the `<RichText>` component. It outputs `<p></p>` when it's empty, just like this Rich Text control does:

<img width="988" alt="empty-paragraph-block" src="https://user-images.githubusercontent.com/4063887/66356531-db645b00-e930-11e9-80de-d5a002f8af93.png">

This is definitely debatable, though. The paragraph block isn't exactly like this.


>The logic above says "After removing all the HTML, if I'm left with an empty string, return an empty string"

Ah, `esc_html()` doesn't usually remove HTML, it just escapes it. 

For example, `esc_html( '<div></div>' )` returns `&lt;div&gt;&lt;/div&gt;`

Reply:
Maybe [strip_tags()](https://www.php.net/manual/en/function.strip-tags.php)

Reply:
Ah – of course! Sorry, I did mean strip_tags. Even better, though: [`wp_strip_all_tags`](https://developer.wordpress.org/reference/functions/wp_strip_all_tags/).

Reply:
Sure, I thought that's what you meant.

Reply:
Though it's unlikely, it's possible that a Rich Text field has only an image, in which case it would be stripped entirely, as there's no text other than the markup:

<img width="855" alt="image-in-rich-text" src="https://user-images.githubusercontent.com/4063887/66361071-00fa6000-e943-11e9-9191-0a67cf4239c3.png">

Reply:
Though https://github.com/getblocklab/block-lab/pull/444/commits/a65299c07cbe1432f27b760aa511c2dcc12b4ef6 changes this to return `''` if the Rich Text field is empty (if it saves `<p></p>`).

Reply:
Aah! Great pickup on the image!

Reply:
Thanks, it's kind of an edge case. Otherwise, `strip_tags()` would work really well.